src/SimulationRunner.cc

```diff
@@ -155,6 +159,21 @@ SimulationRunner::SimulationRunner(const sdf::World *_world,
   // Load the active levels
   this->levelMgr->UpdateLevelsState();
 
+  // Load any additional plugins from the Server Configuration
+  this->LoadServerPlugins(this->serverConfig.Plugins());
+
+  // If we have reached this point and no systems have been loaded, then load
+  // a default set of systems.
+  if (this->systems.empty() && this->pendingSystems.empty())
+  {
+    ignmsg << "No systems loaded from SDF, loading defaults" << std::endl;
+    bool isPlayback = !this->serverConfig.LogPlaybackPath().empty();
+    auto plugins = ignition::gazebo::loadPluginInfo(isPlayback);
+    this->LoadServerPlugins(plugins);
+  }
+
+  this->LoadLoggingPlugins(this->serverConfig);
```

Comment:
The level manager will load all plugins attached to the world on the SDF file above at around line 120. So I think any modification we want to do to the log path must be done before that so we can overwrite the values on the SDF.

I don't think we can escape the logic in `ServerPrivate::AddRecordPlugin`.